pkgs/build-support/appimage/appimage-exec.sh

```diff
@@ -0,0 +1,142 @@
+#!@shell@
+if [ -n "$DEBUG" ] ; then
+  set -x
+fi
+
+PATH="@path@:$PATH"
+apprun_opt=true
+
+#DEBUG=0
+
+# src : AppImage
+# dest : let's unpack() create the directory
+unpack() {
+  local src=$1
+  local out=$2
+  local appimageSignature=""
+  local appimageType=0
+
+  # https://github.com/AppImage/libappimage/blob/ca8d4b53bed5cbc0f3d0398e30806e0d3adeaaab/src/libappimage/utils/MagicBytesChecker.cpp#L45-L63
+  eval "$(r2 "$src" -nn -Nqc "p8j 3 @ 8" |
+    jq -r '{appimageSignature: (.[:-1]|implode), appimageType: .[-1]}|
+      @sh "appimageSignature=\(.appimageSignature) appimageType=\(.appimageType)"')"
+
+  # check AppImage signature
+  if [[ "$appimageSignature" != "AI" ]]; then
+    echo "Not an appimage."
+    exit
+  fi
+
+  case "$appimageType" in
+    1 ) echo "Uncompress $(basename "$src") of type $appimageType."
+        mkdir "$out"
+        pv "$src" | bsdtar -x -C "$out" -f -
+        ;;
+    2)
+        # This method avoid issues with non executable appimages,
+        # non-native packer, packer patching and squashfs-root destination prefix.
+
+        # multiarch offset one-liner using same method as AppImage
+        # see https://gist.github.com/probonopd/a490ba3401b5ef7b881d5e603fa20c93
+        offset=$(r2 "$src" -nn -Nqc "pfj.elf_header @ 0" |\
```

Comment:
Also not having a radare2 as an example would be nice, it might be good enough for now, given that getting this sort of information might be hard to extract using other tools.

Reply:
This was already discussed and merge on another PR.